src/screens/videos/index.js

```diff
@@ -1,46 +1,42 @@
 import { useState } from "react";
 import { VideosDesktop } from "./desktop";
 import { VideosMobile } from "./mobile";
+import useIsMobile from "../../hooks/is-mobile";
 import "./videos.css";
 
 export function Videos() {
-  const [videoPlayers, setVideoPlayers] = useState({});
+  const isMobile = useIsMobile();
+
   const [playingVideoId, setPlayingVideoId] = useState(null);
+  const [activePlayer, setActivePlayer] = useState(null);
 
-  const onReady = (videoId, event) => {
-    setVideoPlayers({ ...videoPlayers, [videoId]: event.target });
-  };
+  const play = (event) => {
+    setActivePlayer(event.target);
 
-  const stopAll = () => {
-    document.querySelectorAll(".yt-iframe").forEach((iframe) => {
-      // @ts-ignore
-      iframe.contentWindow.postMessage(
-        JSON.stringify({ event: "command", func: "stopVideo" }),
-        "*"
-      );
-    });
+    // For some reason, without this, it only works the first time (at least on FF)
+    setTimeout(() => {
+      event.target.playVideo();
+    }, 500);
   };
 
-  const play = (videoId) => {
-    setPlayingVideoId(videoId);
-    stopAll();
-    videoPlayers[videoId]?.playVideo();
+  const stop = () => {
+    activePlayer?.stopVideo();
   };
 
-  return (
-    <>
-      <VideosDesktop
-        className="hidden md:block"
-        playingVideoId={playingVideoId}
-        onReady={onReady}
-        onPlay={play}
-      />
-      <VideosMobile
-        className="md:hidden"
-        playingVideoId={playingVideoId}
-        onReady={onReady}
-        onPlay={play}
-      />
-    </>
+  return isMobile ? (
```

Comment:
Why is this `isMobile` necessary?

Reply:
This avoids loading two YouTube players (mobile + desktop ones), which takes unnecessary computing time (and generates some random console errors, sadly). I think them being hidden doesn't prevent them from being loaded, but I may be mistaken.

(Btw we can't have two elements with the same ID even if one of them is hidden, but that would be fixed by having the ID on the parent component.)